### PR TITLE
Unit tests for gui/network.go

### PR DESCRIPTION
**Pull request by Vavilen**
- /network/connection
- /network/connections
- /network/defaultConnections
- /network/connections/trust
- /network/connections/exchange

### CODE REVIEW

**Comment by Vavilen** [comment]
#998

File: src/gui/network_test.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+			rr := httptest.NewRecorder()
+			handler := NewServerMux(configuredHost, ".", gateway, &CSRFStore{})
+			handler.ServeHTTP(rr, req)
+
+			status := rr.Code
+			require.Equal(t, tc.status, status, "case: %s, handler returned wrong status code: got `%v` want `%v`", tc.name, status, tc.status)
+
+			if status != http.StatusOK {
+				require.Equal(t, tc.err, strings.TrimSpace(rr.Body.String()), "case: %s, handler returned wrong error message: got `%v`| %s, want `%v`",
+					tc.name, strings.TrimSpace(rr.Body.String()), status, tc.err)
```
Comment (therealssj) [inline src/gui/network_test.go]: it should be 
`handler returned wrong error message: got `%v`| %d, want `%v`
`status` is an `int`

File: src/gui/network_test.go [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
```
Comment (therealssj) [inline src/gui/network_test.go]: same as above

File: src/gui/network_test.go [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
```
Comment (therealssj) [inline src/gui/network_test.go]: same as above

File: src/gui/network_test.go [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
```
Comment (therealssj) [inline src/gui/network_test.go]: same as above

File: src/gui/network_test.go [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
```
Comment (therealssj) [inline src/gui/network_test.go]: same as above

File: src/gui/network_test.go (hunk 1 above)
Comment (Vavilen) [reply src/gui/network_test.go]: @therealssj, fixed.

File: src/gui/network_test.go [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+			gatewayGetConnectionResult: nil,
+			result: nil,
+		},
+		{
+			name:   "200",
+			method: http.MethodGet,
+			status: http.StatusOK,
+			err:    "",
+			addr:   "addr",
+			gatewayGetConnectionResult: &daemon.Connection{},
```
Comment (iketheadore) [inline src/gui/network_test.go]: The gatewayGetConnectionResult is empty, can we use some data for mocking.

**Comment by gz-c** [comment]
@Vavilen needs changes

**Comment by gz-c** [comment]
#1018

**Pull request merged**